website/projects/awssync.py

```diff
@@ -36,3 +219,499 @@ def create_aws_organization(self):
+        """
+        Generate the list of users that are registered on the GiPhouse website, but are not yet invited for AWS.
+
+        This includes their ID and email address, to be able to put users in the correct AWS organization later.
+        """
+        sync_list = [x for x in giphouse_data if x not in aws_data]
+        return sync_list
+
+    def create_scp_policy(self, policy_name, policy_description, policy_content):
+        """
+        Create an SCP policy.
+
+        :param policy_name: The policy name.
+        :param policy_description: The policy description.
+        :param policy_content: The policy configuration as a dictionary.
+        The policy is automatically converted to JSON format, including escaped quotation marks.
+        :return: Details of newly created policy as a dict on success and NoneType object otherwise.
+        """
+        client = boto3.client("organizations")
+        try:
+            response = client.create_policy(
+                Content=json.dumps(policy_content),
+                Description=policy_description,
+                Name=policy_name,
+                Type="SERVICE_CONTROL_POLICY",
+            )
+        except ClientError as error:
+            self.fail = True
+            self.logger.error("Something went wrong creating an SCP policy.")
+            self.logger.error(error)
+        else:
+            return response["Policy"]
+
+    def attach_scp_policy(self, policy_id, target_id):
+        """
+        Attaches an SCP policy to a target (root, OU, or member account).
+
+        :param policy_id: The ID of the policy to be attached.
+        :param target_id: The ID of the target root, OU, or member account.
+        """
+        client = boto3.client("organizations")
+        try:
+            client.attach_policy(PolicyId=policy_id, TargetId=target_id)
+        except ClientError as error:
+            if error.response["Error"]["Code"] != "DuplicatePolicyAttachmentException":
+                self.fail = True
+            self.logger.error("Something went wrong attaching an SCP policy to a target.")
+            self.logger.debug(f"{error}")
+            self.logger.debug(f"{error.response}")
+
+    def check_aws_api_connection(self):
+        """
+        Check whether boto3 can connect to AWS API with current credentials.
+
+        :returns: First tuple element always exists and indicates success.
+                  Second tuple element is contains information about the entity
+                  who made the successful API call and None otherwise.
+        """
+        client_sts = boto3.client("sts")
+        try:
+            caller_identity_info = client_sts.get_caller_identity()
+        except (NoCredentialsError, ClientError) as error:
+            self.logger.info("Establishing AWS API connection failed.")
+            self.logger.debug(error)
+            return False, None
+        else:
```

Comment:
Isn't this else unnecessary? If it raises an error then you return false and none, if no error is raised then it will go do the next line after the try catch.